packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -809,14 +954,26 @@ const fieldProxyHandler: AdaptingProxyHandler<FieldProxyTarget, EditableField> =
                 return target;
             case Symbol.iterator:
                 return target[Symbol.iterator].bind(target);
+            case arrayLikeMarkerSymbol:
+                return true;
             default:
         }
         return undefined;
     },
-    set: (target: FieldProxyTarget, key: string, value: unknown, receiver: unknown): boolean => {
-        assert(keyIsValidIndex(key, target.length), 0x459 /* The node does not exist. */);
-        const node = target.proxifyNode(Number(key), false);
-        node[valueSymbol] = value;
+    set: (
+        target: FieldProxyTarget,
+        key: string,
+        value: ContextuallyTypedNodeData,
+        receiver: unknown,
+    ): boolean => {
+        const cursor = singleMapTreeCursor(
+            applyTypesFromContext(target.context.schema, target.fieldSchema.types, value),
+        );
+        if (keyIsValidIndex(key, target.length)) {
+            target.replaceNodes(Number(key), cursor, 1);
+        } else {
+            target.insertNodes(Number(key), cursor);
```

Comment:
Maybe we should assert that `Number(key)` is not NaN.

Reply:
`keyIsValidIndex` is a cheap way to check if there might be a node at the given index. Key checks are implemented inside the target. I'll add some comments to avoid confusion and also some to the implementation, since currently we don't check if the node really exists while replacing it, which might be desired, but could also be costly.